### PR TITLE
case-insensitive metadata

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

File: sdk/storage/CMakeLists.txt [hunk 1]
```diff
@@ -35,5 +35,5 @@ endif()
 
 add_subdirectory(azure-storage-common)
 add_subdirectory(azure-storage-blobs)
-add_subdirectory(azure-storage-files-datalake)
-add_subdirectory(azure-storage-files-shares)
+# add_subdirectory(azure-storage-files-datalake)
+# add_subdirectory(azure-storage-files-shares)
```
Comment (ahsonkhan) [inline sdk/storage/CMakeLists.txt:39]: friendly reminder: remember to uncomment these before committing.

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp [hunk 2]
```diff
@@ -291,13 +291,11 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       Azure::Core::Nullable<std::string> NextAllowedQuotaDowngradeTime;
       std::string DeletedTime;
       int32_t RemainingRetentionDays = int32_t();
-      Models::LeaseStatusType LeaseStatus;
-      Models::LeaseStateType LeaseState;
-      Models::LeaseDurationType LeaseDuration;
+      Models::LeaseStatusType LeaseStatus = Models::LeaseStatusType::Unknown;
+      Models::LeaseStateType LeaseState = Models::LeaseStateType::Unknown;
+      Models::LeaseDurationType LeaseDuration = Models::LeaseDurationType::Unknown;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/protocol/share_rest_client.hpp:296]: woh, only do one thing in a PR.

**Review by vinjiang — APPROVED** [review]
Better to keep PR on single purposes.

**Pull request merged**